x-pack/plugins/apm/public/components/app/service_overview/service_overview_transactions_table/index.tsx

```diff
@@ -258,6 +195,127 @@ export function ServiceOverviewTransactionsTable(props: Props) {
+  const { serviceName } = props;
+  const { transactionType } = useApmServiceContext();
+  const {
+    uiFilters,
+    urlParams: { start, end, latencyAggregationType },
+  } = useUrlParams();
+
+  const [tableOptions, setTableOptions] = useState<{
+    pageIndex: number;
+    sort: {
+      direction: SortDirection;
+      field: SortField;
+    };
+  }>({
+    pageIndex: 0,
+    sort: DEFAULT_SORT,
+  });
+
+  const {
+    data = {
+      totalItemCount: 0,
+      items: [],
+      tableOptions: {
+        pageIndex: 0,
+        sort: DEFAULT_SORT,
+      },
+    },
+    status,
+  } = useFetcher(() => {
+    if (!start || !end || !latencyAggregationType || !transactionType) {
+      return;
+    }
+
+    return callApmApi({
+      endpoint:
+        'GET /api/apm/services/{serviceName}/transactions/groups/overview',
+      params: {
+        path: { serviceName },
+        query: {
+          start,
+          end,
+          uiFilters: JSON.stringify(uiFilters),
+          numBuckets: 20,
+          sortField: tableOptions.sort.field,
+          sortDirection: tableOptions.sort.direction,
+          transactionType,
+          latencyAggregationType: latencyAggregationType as LatencyAggregationType,
+        },
+      },
+    }).then((response) => {
+      return {
+        items: response.transactionGroups,
+        totalItemCount: response.totalTransactionGroups,
+        tableOptions: {
+          pageIndex: tableOptions.pageIndex,
+          sort: {
+            field: tableOptions.sort.field,
+            direction: tableOptions.sort.direction,
+          },
+        },
+      };
+    });
+  }, [
+    serviceName,
+    start,
+    end,
+    uiFilters,
+    tableOptions.pageIndex,
+    tableOptions.sort.field,
+    tableOptions.sort.direction,
+    transactionType,
+    latencyAggregationType,
+  ]);
+
+  const {
+    items,
+    totalItemCount,
+    tableOptions: { pageIndex, sort },
+  } = data;
+
+  const transactionNames = useMemo(() => items.map(({ name }) => name), [
+    items,
+  ]);
+
+  const {
+    data: transactionsMetricsData,
+    status: transactionsMetricsStatus,
+  } = useFetcher(
+    () => {
+      if (transactionNames.length && start && end && transactionType) {
+        return callApmApi({
+          endpoint:
+            'GET /api/apm/services/{serviceName}/transactions/groups/metrics',
+          params: {
+            path: { serviceName },
+            query: {
+              start,
+              end,
+              uiFilters: JSON.stringify(uiFilters),
+              numBuckets: 20,
+              transactionType,
+              latencyAggregationType: latencyAggregationType as LatencyAggregationType,
+              transactionNames: transactionNames.join(),
+            },
+          },
+        });
+      }
+    },
+    // only fetches metrics when transaction names change
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [transactionNames]
```

Comment:
I'm not sure if this is the most robust way to manage the second call. It's assuming the identity of `items` will only change when the data does, but you have to go through the code line-by-line to ensure that is actually the case. It's also hard for me to figure out what happens when the primary stats for the transaction groups are updated (e.g. by changing the date range), but this call has not finished yet. Will it merge comparison stats for the previous date range with the primary stats for the current date range?

Reply:
> Will it merge comparison stats for the previous date range with the primary stats for the current date range?

Didn't we agree that the current `{serviceName}/transactions/groups/metrics` would return both the current metrics and comparison metrics?

Reply:
Primary stats: 
- list of transaction group names
- avg/p95/p99 for latency, avg for throughput/error rate
- impact

Comparison stats:
- spark plots
- impact

What do you mean with "current metrics" and "comparison metrics"?